ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/ReadsKeySets.hs

```diff
@@ -153,3 +157,22 @@ withHydratedLedgerState st dbch urs f =
       .   withLedgerTables st
       <$> forwardTableKeySets dbch urs
 
+{-------------------------------------------------------------------------------
+  Reading tables for specific points, to be used by the mempool
+-------------------------------------------------------------------------------}
+
+-- | The requested point is not found on the ledger db
+data PointNotFound blk = PointNotFound !(Point blk) deriving (Eq, Show)
+
+-- | Read and forward the values up to the tip of the given ledger db. Returns
+-- Left if the anchor moved.
```

Comment:
Perhaps we should explain what the value inside the `Left` means (what do `fst` and `snd` represent?).